### ISSUE TITLE
Symk not working.

### ISSUE THREAD

**Issue by MFaisalZaki** [aiplan4eu/up-symk#1]
I'm trying to solve the rover problem using `Symk`; however, it throws an error in the `preprocess` phase:
<img width="1188" alt="Screenshot 2023-09-20 at 10 10 41 PM" src="https://github.com/aiplan4eu/up-symk/assets/9129265/139cd362-496b-4b8c-a8bc-0e210a866f73">

Here is the code:
```
from collections import defaultdict
from unified_planning.io import PDDLReader
from unified_planning.shortcuts import *

import up_symk

planning_task = PDDLReader().parse_problem(`domain.pddl`, `problem.pddl`)

_symk_cfg = defaultdict(dict)
_symk_cfg['number_of_plans'] = 5
_symk_cfg['plan_cost_bound'] = 1

planners = [('symk', _symk_cfg)]
planners_resutls = []
for plannername, args in planners:
    with OneshotPlanner(name=plannername, params=args) as planner:
        result = planner.solve(planning_task)
        planners_resutls.append((plannername, result))

```

**Comment by speckdavid** [comment]
Thanks for the report. Can you send me the pddl files or are these official IPC files?

**Comment by MFaisalZaki** [comment]
Sure here are the files,
[pddl-files.zip](https://github.com/aiplan4eu/up-symk/files/12682824/pddl-files.zip), Sorry, I forgot to upload it.

**Comment by speckdavid** [comment]
Thanks. I just ran it on my machine with a clean Python 3.11 environment installed with the latest pypi packages from unified-planning and up-symk. Everything works for me, so I cannot reproduce the error, yet. 

Can you check if you have the latest packages installed (e.g. with pip list)?
- unified-planning 1.0.0
- up-symk 0.0.5

If the latest version of the packages are installed and the error still occurs, it would be helpful to see the output of symk.

Running "result = planner.solve(planning_task, output_stream=sys.stdout)" will print the output of the planner, which might help to diagnose the error.

**Comment by MFaisalZaki** [comment]
I have reinstalled both packages (symk from the latest commit), and I got the same error. Here is the output of `Symk`. Apparently, there is an issue in the translation phase: `Failed to match magic word 'begin_rule'`.
<img width="1188" alt="Screenshot 2023-09-21 at 8 03 24 PM" src="https://github.com/aiplan4eu/up-symk/assets/9129265/7d3dc264-8d02-45fc-97d0-3eccf3523b5e">

**Comment by speckdavid** [comment]
Mhh, not quite clear to me what is happening here. It looks like the preprocessor is not accepting the output.sas. Normally, if the planner stops at this point, the output.sas should not be deleted from the folder and can be inspected. Could you maybe upload the generated output.sas?

Also, would you mind trying to uninstall the local installation of up-symk (`pip uninstall up-symk`) and install it via pypi with `pip install --no-cache-dir -v up-symk`? (Run this command in a folder that does not have a folder called up-symk.) If it worked correctly, pip freeze should show up-symk==0.0.5. 

Another point that would be interesting is whether the one shot planner "fast-downward" works?

**Comment by speckdavid** [comment]
I think I have identified the issue and have released a new version of up-symk==0.0.6 which will hopefully fix the problem. Can you please install the new package and see if everything works now?

By the way, some comments on the code you provided. At the moment symk is called as a Oneshot planner, which produces only a single plan even if the number_of_plans is specified. I would recommend calling symk as an Anytime planner, like this:

```
from collections import defaultdict
from unified_planning.io import PDDLReader
from unified_planning.shortcuts import *

import up_symk

planning_task = PDDLReader().parse_problem('domain.pddl', 'problem.pddl')

_symk_cfg = defaultdict(dict)
_symk_cfg['number_of_plans'] = 3
_symk_cfg['plan_cost_bound'] = 1000

plans = []

with AnytimePlanner(name='symk', params=_symk_cfg) as planner:
    for i, result in enumerate(planner.get_solutions(planning_task)): # output_stream=sys.stdout): 
        if result.status == up.engines.PlanGenerationResultStatus.INTERMEDIATE:
            plans.append(result.plan)
            print(f"Plan {i+1}: {result.plan}.")
            print()
        elif result.status in unified_planning.engines.results.POSITIVE_OUTCOMES:
            print(f"{planner.name} found {len(plans)}.")
        elif result.status not in unified_planning.engines.results.POSITIVE_OUTCOMES:
            print("No plan found.") 
```

A more detailed example you can be found in this [notebook](https://github.com/aiplan4eu/up-symk/blob/master/notebooks/symk_usage.ipynb). 

Finally, in the 1.0.0 release of the unified-planning library, there was a problem with the plan parser. So I would recommend installing unified-planning and symk the following to avoid problems until a new version of the unified-planning library is released.

```
git clone git@github.com:aiplan4eu/unified-planning.git
pip install unified-planning/
pip install up-symk
```

**Comment by MFaisalZaki** [comment]
Thanks for the updates, @speckdavid. I have tested them, and they are working perfectly.
